aip/general/0213.md

```diff
@@ -18,24 +18,105 @@ into taking major version changes or even lead to dependency conflicts.
 
-APIs **should not** directly depend on protos defined in other APIs. Instead,
+APIs **must not** directly depend on protos defined in other APIs. Instead,
 they should copy and paste the applicable messages into their own API. When
-doing so, APIs **should** keep the field names and numbers the same.
-
-## Existing common protos
+doing so, APIs **should** keep the field names and numbers the same. To refer
+to resources from other APIs, specify [resource names](https://aip.dev/122)
+using `string` fields.
+
+### Rationale
+
+When one API depends on protos defined by another API, this introduces
+uncertainty in terms of customer-expected behavior and client library
+dependency management. Suppose `google.cloud.library.v1` depends on
+the protos (rather than abstract resources) in `google.cloud.movies.v2`. Any
+change to `google.cloud.movies.v2` can cause problems. For example:
+
+- If a field is added to a message in `google.cloud.movies.v2`, should
+  customers using `google.cloud.library.v1` expect to see it? If so, how soon
+  after the field has been added? What about other API changes?
+- If the whole major version is deprecated (typically after v3 has been
+  released), does that mean `google.cloud.library.v1` has to change to use
+  `google.cloud.movies.v3`, and if so, does that require a new major version for
+  the library API as well?
+- How should client library versioning reflect changes to dependent APIs?
+
+Keeping APIs isolated from each other, with a limited set of common protos which
+are maintained in a highly disciplined way, reduces a lot of the issues with
+dependencies.
+
+## Organization-specific common protos
+
+Organizations often have many different APIs all centered around the same set of
+concepts which are organization-specific. Where it is safe to share a single
+representation across multiple APIs, this can make it much easier for clients to
+interact with multiple APIs. Concept-specific client code can be written once,
+and messages can be used from the response of one API in the request of another
+without clunk copying, for example.
+
+This benefit comes with significant restrictions and limitations, however, and
+should not be attempted lightly.
+
+When a common proto is desirable but not genuinely organization-specific, it
+**should** be introduced into a global common proto package instead, even if an
+organization-specific common proto package has already been created.
+
+## Defining an organization-specific common proto package
+
+Any organization wishing to define a common proto package should first consider
+whether they really need to. Introducing a common proto package has costs in
+terms of client library packaging, customer documentation, and general
+discipline in following the strict limitations. If a single simple message is
+going to be shared between only a few APIs, the benefits may not be worth the
+costs.
+
+If many APIs can be identified that *definitely* need the same concept,
```

Comment:
make this a bulleted list? easier to consume.

Reply:
(See above.)